src/vs/vscode.d.ts

```diff
@@ -2719,6 +2719,57 @@ declare module 'vscode' {
 	}
 
 	/**
+	 * Contains additional information about the context in which
+	 * [completion provider](#CompletionItemProvider.provideCompletionItems) is triggered.
+	 */
+	export interface CompletionContext {
+		/**
+		 * How the completion was triggered.
+		 */
+		readonly trigger: 'auto' | 'manual';
+
+		/**
+		 * Character that triggered the completion item provider.
+		 *
+		 * `undefined` if provider was not triggered by a character.
+		 */
+		readonly triggerCharacter?: string;
```

Comment:
We should also point out that this is purely convenience as the trigger character is already in the document and get be looked up at the provided position.